db/migrate/20191108135108_add_service_credential_tables.rb

```diff
@@ -0,0 +1,58 @@
+class AddServiceCredentialTables < ActiveRecord::Migration[5.2]
+  def change
+    create_table :service_credential_types, :id => :bigserial, :force => :cascade do |t|
+      t.references :tenant, :index => true, :null => false, :foreign_key => { :on_delete => :cascade }
+      t.references :source, :index => true, :null => false, :foreign_key => { :on_delete => :cascade }
+
+      t.string :source_ref, :null => false
+      t.string :name
+      t.text :description
+
+      t.jsonb :extra # For credentials input fields
```

Comment:
not sure about this, AnsibleTower has several booleans for this, but don't know about other provider types (if applicable)